tests/components/element-card/element-card.spec.tsx

```diff
@@ -0,0 +1,105 @@
+import { fireEvent, render, screen } from "@testing-library/react";
+import "jest-canvas-mock";
+import { ElementCard } from "src/components";
+import { EElementType, TElement } from "src/context-providers";
+import { ELEMENT_BUTTON_LABELS } from "src/data/elements-data";
+import { TestHelper, mockBuilderState } from "src/util/test-helper";
+
+const mockElement = {
+    internalId: "mock123",
+    type: EElementType.EMAIL,
+    id: "mockElement",
+    required: false,
+    label: ELEMENT_BUTTON_LABELS[EElementType.EMAIL],
+};
+
+const mockOnClick = jest.fn();
+
+describe("element-card", () => {
+    afterEach(() => {
+        jest.restoreAllMocks();
+        jest.resetAllMocks();
+    });
+
+    describe("Testing onClick", () => {
+        it("should be in a focused state and show the edit element panel when clicked on", () => {
+            renderComponent({ element: mockElement, onClick: mockOnClick }, {});
+            fireEvent.click(getElementCard() as HTMLElement);
+            expect(mockOnClick).toHaveBeenCalled();
+        });
+    });
+
+    describe("Testing rendering of duplicate & delete button based on isFocused", () => {
+        it("should not show the duplicate and delete buttons when in normal state", () => {
+            renderComponent({ element: mockElement });
+            expect(getDuplicateButton(true)).not.toBeInTheDocument();
+            expect(getDeleteButton(true)).not.toBeInTheDocument();
+        });
+
+        it("should show the duplicate and delete buttons when in focused state", () => {
+            renderComponent(
+                { element: mockElement },
+                {
+                    builderContext: {
+                        ...mockBuilderState,
+                        focusedElement: { element: mockElement },
+                    },
+                }
+            );
+            expect(getDuplicateButton()).toBeInTheDocument();
+            expect(getDeleteButton()).toBeInTheDocument();
+        });
+    });
+
+    describe("Testing disabling of the duplicate button", () => {
```

Comment:
don't see a need to wrap the test in a test suite since there is only one test here